lib/routes/routeBackbeat.js

```diff
@@ -497,6 +497,11 @@ function putMetadata(request, response, bucketInfo, objMd, log, callback) {
                     // If no new versioned objects are added for given object(s), they look like
                     // standalone master keys.
                     versioning = false;
+                    // Delete the version id from the version metadata payload to prevent issues
+                    // with creating a non-version object (versioning set to false) that includes a version id.
+                    // For example, this version ID might come from a null version of a suspended bucket being
+                    // replicated to this bucket.
+                    delete omVal.versionId;
```

Comment:
It would look clearer to me if we do this after `omVal.isNull = true;` above, if that doesn't change the core logic of the fix. But I am not sure I understand fully the logic.

Reply:
It is indeed cleaner to have this check after `omVal.isNull = true;` and indeed does not change the core logic since putMetadata to **insert** a new metadata key is only used for replication by S3C (Artesca uses Cloud provider SDKs) that only supports **versioned bucket**.